svg-icons-builder/schema.json

```diff
@@ -1,77 +1,217 @@
+                "default": "myIcon"
+              },
+              "interfaceName": {
+                "type": "string",
+                "description": "Name for the generated interface.",
+                "default": "MyIcon"
+              }
+            },
+            "required": [
+              "fileName",
+              "typeName",
+              "generateType",
+              "generateTypeObject",
+              "generateCompleteIconSet",
+              "prefix",
+              "interfaceName"
+            ]
+          }
+        ]
+      }
     },
-    "fileName": {
-      "type": "string",
-      "description": "File name of the generated constants file.",
-      "default": "my-icons"
-    },
-    "typeName": {
-      "type": "string",
-      "description": "Name of the generated type.",
-      "default": "myIcons"
-    },
-    "generateType": {
-      "type": "boolean",
-      "description": "Prevent generating enumeration type.",
-      "default": false
-    },
-    "generateTypeObject": {
-      "type": "boolean",
-      "description": "Generate type object.",
-      "default": false
-    },
-    "generateCompleteIconSet": {
-      "type": "boolean",
-      "description": "TODO: is this the same as: exportCompleteIconSet",
-      "default": false
-    },
-    "exportCompleteIconSet": {
-      "type": "boolean",
-      "description": "Specifies if the complete icon set should be exported or not (can be very handy for showcases)",
-      "default": false
-    },
-    "prefix": {
-      "type": "string",
-      "description": "Prefix for the generated svg constants.",
-      "default": "myIcon"
-    },
-    "interfaceName": {
-      "type": "string",
-      "description": "Name for the generated interface.",
-      "default": "MyIcon"
-    },
-    "srcFiles": {
-      "type": "array",
-      "description": "Input files matching the given filename pattern.",
-      "items": [
-        {
-          "type": "string"
-        }
-      ],
-      "default": "[\"/*.svg\"]"
-    },
-    "outputDirectory": {
-      "type": "string",
-      "description": "Name of the output directory.",
-      "default": "\"./dist\""
+    {
+      "if": {
+        "properties": { "conversionType": { "const": "object" } }
+      },
+      "then": {
+        "allOf": [
+          { "$ref": "#/definitions/baseOptions" },
+          {
+            "properties": {
+              "fileName": {
+                "type": "string",
+                "description": "File name of the generated constants file.",
+                "default": "my-icons"
+              },
+              "obejctName": {
+                "type": "string",
+                "description": "name of the exported const - if nothing is set - default export will be used",
+                "default": "icons"
+              }
+            },
+            "required": ["fileName", "objectName"]
+          }
+        ]
+      }
     },
-    "svgoConfig": {
+    {
+      "if": {
```

Comment:
Very cool - I didn't know that you can use `if` `then` statements in `schema.json`

Reply:
I didn't either! I was trying to re-use fields and found this. The nice tool online to test schemas does not yet support this but working in small bits worked.